src/examples/generation.ts

```diff
@@ -0,0 +1,100 @@
+import { Armature } from '../armature/Armature';
+import { Node, Point } from '../armature/Node';
+import { genSphere } from '../geometry/Sphere';
+import { Light } from '../renderer/interfaces/Light';
+import { Renderer } from '../renderer/Renderer';
+
+import { mat4, quat, vec3 } from 'gl-matrix';
+import { flatMap } from 'lodash';
+
+const light1: Light = {
+    lightPosition: [10, 10, 10],
+    lightColor: [0.3, 0.3, 0.3],
+    lightIntensity: 256
+};
+const light2: Light = {
+    lightPosition: [700, 500, 50],
+    lightColor: [0.3, 0.3, 0.3],
+    lightIntensity: 100
+};
+
+const renderer: Renderer = new Renderer(800, 600, 2, vec3.fromValues(0.2, 0.2, 0.2));
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+// Step 1: create geometry
+///////////////////////////////////////////////////////////////////////////////////////////////////
+
+// Add lights to the renderer
+renderer.addLight(light1);
+renderer.addLight(light2);
+
+const leafSphere = genSphere();
+leafSphere.colors = Float32Array.from(flatMap(leafSphere.vertices, () => [0.8, 1, 0.8]));
```

Comment:
Do we want people using our library to have to import something like lodash to use the `flatMap` feature, or should we provide methods that could do the flattening for them internally so they don't need to import other libraries?

Reply:
I realize this is a bit of a nit, but I'd like it if our examples could just use vanilla ES6 or TypeScript using our library. So eventually it would be nice to handle the `gl-matrix` stuff in the library too. This is beyond the scope of the PR, but I think it would be good to think about in the future. Thoughts?

Reply:
I'd like to consider this at a later stage in the project! I like the idea 😄, especially when we're targeting beginners but I want to avoid possible bloat in our library as of right now. Wdyt?

Reply:
Yeah, sounds like a good idea to me @aneedalie!

Reply:
Agreed, end users should never have to directly deal with glmatrix. Paul is working on a pr to remove the need for vec3 in client code right now

Reply:
Also the flatmap will go away once we add colour to our geometry functions. This is a placeholder while we don't have any API for it